res/controllers/Hercules-DJ-Console-4-Mx-scripts.js

```diff
@@ -0,0 +1,622 @@
+};
+
+//AutoDJ is activated or deactivated.
+Hercules4Mx.onAutoDJ = function(value, group, control) {
+    midi.sendShortMsg(Hercules4Mx.NOnC1, 0x3C, (value) ? 0x7F : 0x00);
+};
+
+// AutoDJ fade to next is pressed. (seems it isn't called when the fading is done 
+// automatically by AutoDJ instead of button pressed
+Hercules4Mx.onAutoDJFade = function(value, group, control) {
+    //Flashing led to indicate fading
+    midi.sendShortMsg(Hercules4Mx.NOnC1, 0x7C, 0x7F);
+    if (Hercules4Mx.autoDJfadingId !== 0) {
+        //Ensure the timer is off.
+        //This is a safety measure in case the button is pressed again within the 5 second delay.
+        engine.stopTimer(Hercules4Mx.autoDJfadingId);
+        Hercules4Mx.autoDJfadingId = 0;
+    }
+    //After 5 seconds, restore non-flashing led. It would be perfect if autoDJFade was triggered also
+    //when the fading ends, but right now it seems this is not possible. Also, it doesn't seem to be
+    //an option to get the duration of the fading, that's why i simply put there 5 seconds.
+    Hercules4Mx.autoDJfadingId = engine.beginTimer(5000, "Hercules4Mx.onAutoDJFadeOff", true);
+};
+Hercules4Mx.onAutoDJFadeOff = function() {
+    midi.sendShortMsg(Hercules4Mx.NOnC1, 0x7C, 0x00);
+};
+
+///////////////////////////////////////////////////////////////////
+// --- Actions ----
+//Auto DJ button is pressed. Two functions: enable autoDJ, and Fade to next track.
+//The virtualDJ function is to do a fade to next in interactive mode (Mixxx fade to next requires autoDJ)
+//The would then be determined by the beats per minute.
+Hercules4Mx.autoDJButton = function(midichan, control, value) {
+    if (value) {
+        if (engine.getParameter("[AutoDJ]", "enabled") === 0) {
+            //If not enable, enable autoDJ
+            engine.setParameter("[AutoDJ]", "enabled", 1);
+        } else {
+            //If already in autodj, do a crossfade to next track
+            engine.setParameter("[AutoDJ]", "fade_now", 1);
+        }
+    }
+};
+
+//Stop button is pressed in a deck.
+Hercules4Mx.stopButton = function(midichan, control, value, status, group) {
+    if (value) {
+        engine.setParameter(group, "start_stop", 1);
+
+        var stop1 = engine.getParameter("[Channel1]", "stop");
+        var stop2 = engine.getParameter("[Channel2]", "stop");
+        var autodj = engine.getParameter("[AutoDJ]", "enabled");
+        //Note: autodj only acts on decks 1 and 2, so not checking decks 3 and 4.
+        if (stop1 && stop2 && autodj) {
+            //If autoDJ enabled and all decks are stopped. disable autoDJ.
+            engine.setParameter("[AutoDJ]", "enabled", 0);
+        }
+    }
+};
+
+//Advanced navigation mode. 
+Hercules4Mx.navigation = function(midichan, control, value, status, group) {
+    if (value) {
+        if (control === 0x3E) {
+            //FILES
+            if (Hercules4Mx.navigationStatus.sidebar === 0 &&
+                engine.getParameter("[AutoDJ]", "enabled") === 1) {
```

Comment:
The spacing here is confusing. This line is indented the same as the line below inside the if block. Maybe removing one space so "engine" aligns with the "(" would help. Perhaps it would make most sense to have one long line with both conditions.